tensorflow/python/framework/test_util.py

```diff
@@ -1802,6 +1803,12 @@ def decorated(self, *args, **kwargs):
   return disable_test_impl
 
 
+# The description is just for documentation purposes.
+def disable_nonAVX512f(description):  # pylint: disable=unused-argument
```

Comment:
We would still want to run CUDA tests if CPU has CUDA but doesn't have AVX-512f. Maybe it's better to add a condition to `_maybe_skip` instead:
https://github.com/tensorflow/tensorflow/blob/00476037681d6e10f8c0bca83251551134b169f8/tensorflow/python/grappler/auto_mixed_precision_test.py#L372-L373

Reply:
Hi Penporn, Sorry for the delay, I was in vacation. Did you mean to use _maybe_skip from tensorflow/python/grappler/auto_mixed_precision_test.py. 
or you are suggesting to add "cuda" condition here like below from _maybe_skip : 
  def _maybe_skip(self, mode):
    if mode == 'cuda' and not test.is_gpu_available(cuda_only=True):
      self.skipTest('No GPU is available')
    if mode == 'mkl' and not test_util.IsMklEnabled():
      self.skipTest('MKL is not enabled')

Reply:
Hi Penporn, Sorry for the delay, I was in vacation. Did you mean to use _maybe_skip from tensorflow/python/grappler/auto_mixed_precision_test.py.
or you are suggesting to add "cuda" condition here like below from _maybe_skip :
def _maybe_skip(self, mode):
if mode == 'cuda' and not test.is_gpu_available(cuda_only=True):
self.skipTest('No GPU is available')
if mode == 'mkl' and not test_util.IsMklEnabled():
self.skipTest('MKL is not enabled')

Reply:
If you would like me to add lines inside "_maybe_skip" function, I am guessing if mode is 'cuda' and cuda is available , the test should run (no matter if avx-512f cpu or not?). Is it possible for you to clarify little more?

Reply:
Hi Mazhar, Sorry for my late reply as well! I'm very far behind on github notifications lately.
I was thinking of the first option (use `_maybe_skip` and only modify `auto_mixed_precision_test.py`). But if you expect this to be used somewhere else as well, we can continue with `disable_nonAVX512f` (but not disable it for CUDA tests).

Reply:
I have changed it . Would you please look at it to check if that is the thing you wanted me to do?